### DISCUSSION TITLE
Downloading Prisma engines for Node-API for darwin-arm64 and linux-arm64-openssl-1.0.x [] 0%

### DISCUSSION THREAD

**Discussion by mits87** [prisma/prisma#21764] [Q&A]
### Bug description

Hi, 

I have a problem with Prisma and generating new schema based on below `binaryTargets` (actually the same issue is when I have any other binaryTarget than `native`):

```
generator client {
  provider = "prisma-client-js"
  binaryTargets = ["native", "linux-arm64-openssl-1.0.x"]
}
```

Whenever I execute `prisma generate` command I get this error:

```
Prisma schema loaded from prisma/schema.prisma
> Downloading Prisma engines for Node-API for darwin-arm64 and linux-arm64-openssl-1.0.x [] 0%Error: request to https://binaries.prisma.sh/all_commits/aebc046ce8b88ebbcb45efe31cbe7d06fd6abc0a/linux-arm64-openssl-1.0.x/libquery_engine.so.node.sha256 failed, reason: connect ETIMEDOUT 2606:4700:10::6816:166d:443
error Command failed with exit code 1.
```

When I'm trying to download the file [https://binaries.prisma.sh/all_commits/aebc046ce8b88ebbcb45efe31cbe7d06fd6abc0a/linux-arm64-openssl-1.0.x/libquery_engine.so.node.sha256](https://binaries.prisma.sh/all_commits/aebc046ce8b88ebbcb45efe31cbe7d06fd6abc0a/linux-arm64-openssl-1.0.x/libquery_engine.so.node.sha256) using browser or wget then there is **no problems.** 

Can you help me with that?

### How to reproduce

1. Set `binaryTargets = ["native", "linux-arm64-openssl-1.0.x"]`
2. Execute `prisma generate`
4. See an error `Downloading Prisma engines for Node-API for darwin-arm64 and linux-arm64-openssl-1.0.x [] 0%Error: request to https://binaries.prisma.sh/all_commits/aebc046ce8b88ebbcb45efe31cbe7d06fd6abc0a/linux-arm64-openssl-1.0.x/libquery_engine.so.node.sha256 failed, reason: connect ETIMEDOUT 2606:4700:10::6816:166d:443` 

<img width="1728" alt="image" src="https://github.com/prisma/prisma/assets/605186/5703e5e4-46bf-4479-8653-b4e3cb24b929">

### Prisma information

```prisma
datasource db {
  provider = "mongodb"
  url      = env("PRISMA_DATABASE_URL")
}

generator client {
  provider = "prisma-client-js"
  binaryTargets = ["native", "linux-arm64-openssl-1.0.x"]
}

generator json {
  provider = "prisma-json-types-generator"
}

enum DictionaryType {
  focus
  funds
}
```

### Environment & setup

- OS: osx 14.0 
- Database: MongoDB
- Node.js version: v18.13.0

### Prisma Version

```
prisma                  : 5.5.2
@prisma/client          : 5.5.2
Current platform        : darwin-arm64
Query Engine (Node-API) : libquery-engine aebc046ce8b88ebbcb45efe31cbe7d06fd6abc0a (at node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Schema Engine           : schema-engine-cli aebc046ce8b88ebbcb45efe31cbe7d06fd6abc0a (at node_modules/@prisma/engines/schema-engine-darwin-arm64)
Schema Wasm             : @prisma/prisma-schema-wasm 5.5.1-1.aebc046ce8b88ebbcb45efe31cbe7d06fd6abc0a
Default Engines Hash    : aebc046ce8b88ebbcb45efe31cbe7d06fd6abc0a
Studio                  : 0.494.0
```

**Comment by janpio** [comment]
Something is either blocking the download on your terminal, or your browser is somehow configured differently in a way that makes it work there that has not been applied to your terminal.

Usually switching to a different network already fixes that problem, a different Wi-fi or mobile network connection for example.

**Reply by mits87** [reply]
I can download that file through my browser (pasting url to the browser) and also through CLI like:
```
curl https://binaries.prisma.sh/all_commits/aebc046ce8b88ebbcb45efe31cbe7d06fd6abc0a/linux-arm64-openssl-1.0.x/libquery_engine.so.node.sha256 -o libquery_engine.so.node.sha256
``` 

In both cases works. 
When I use wget then it does not work:
```
--2023-11-03 15:17:26--  https://binaries.prisma.sh/all_commits/aebc046ce8b88ebbcb45efe31cbe7d06fd6abc0a/linux-arm64-openssl-1.0.x/libquery_engine.so.node.sha256
Resolving binaries.prisma.sh (binaries.prisma.sh)... 2606:4700:10::6816:166d, 2606:4700:10::6816:176d, 2606:4700:10::ac43:4ad, ...
Connecting to binaries.prisma.sh (binaries.prisma.sh)|2606:4700:10::6816:166d|:443...
```

I tried to change my DNSes (8.8.8.8, 1.1.1.1 and default) and this didn't help. 
When I switch to mobile network I have the same. 

I'm using my private computer and I'm pretty sure there is no custom / corporate blockers 😮

**Reply by mits87** [reply]
ahhhh I discovered! 
This not gonna work for IPv6 😢  when I switched to IPv4 then it works correctly. 

Is it possible to support IPv6 connection?

**Reply by janpio** [reply]
That should technically be supported, let me investigate.

**Reply by ziperfal** [reply]
Any update about this one? I am not able to run my app locally :(

**Reply by janpio** [reply]
We are not aware of any problems right now.

**Reply by LeeeeeeM** [reply]
the same

**Reply by ludralph** [reply]
Hi @LeeeeeeM 

Please open a new discussion with the relevant details so we can assist in resolving it.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
